examples/WeatherForecast/src/WeatherForecast.Entities/HasId.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+using JetBrains.Annotations;
+
+namespace WeatherForecast.Entities;
+
+[PublicAPI]
+public abstract class HasId
+{
+  [JsonConstructor]
+  public HasId(Guid id)
+  {
+    Id = id;
+  }
+
+  [Key]
+  [JsonPropertyName("id")]
+  public Guid Id { get; }
```

Comment:
readonly?

Reply:
Yes, everything is immutable. Do you see any disadvantages?